packages/ember-glimmer/lib/syntax/dynamic-component.js

```diff
@@ -21,11 +27,27 @@ function dynamicComponentFor(vm) {
 }
 
 export class DynamicComponentSyntax extends StatementSyntax {
-  constructor({ args, templates, parentMeta }) {
+  // for {{component componentName}}
+  static create({ args, templates, parentMeta }) {
+    let definitionArgs = ArgsSyntax.fromPositionalArgs(args.positional.slice(0, 1));
+    let invocationArgs = ArgsSyntax.build(args.positional.slice(1), args.named);
+    return new this({ definitionArgs, args: invocationArgs, templates, parentMeta });
+  }
+
+  // Transforms {{foo.bar}} or {{#foo.bar}}{{/foo.bar}}
+  // into {{component foo.bar}} or {{#component foo.bar}}{{/component}}
+  // with all of it's arguments
+  static fromPathReference({ path, args, templates, parentMeta }) {
+    let positional = ArgsSyntax.fromPositionalArgs(PositionalArgsSyntax.build([GetSyntax.build(path.join('.'))]));
```

Comment:
Once this lands, we should add a `GetSyntax.fromParts` to avoid this join and split dance